components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1566,6 +1567,15 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
                                         ApplicationConstants.LOCAL_IDP_NAME,
                                         lclAuthenticator.getName(),
                                         lclAuthenticator.getDisplayName());
+                            } else {
```

Comment:
Is this else check necessary ?

Reply:
Any authenticator will have an authenticator config right ?
If so when instantiating the config if we made sure these new properties are set do we need to make sure of adding those fields back here for an existing authenticator ?

Shouldn't this go to the point where the authenticator is added to the DB ?
If this is about updating existing entries then it's an ondemand data migration. We need to have a method specific for that so it's easy to read, and clean up later

Reply:
Yeah this is about updating existing db entries. This is improved as suggested.